src/lib/db/client-instance-store.ts

```diff
@@ -182,6 +182,18 @@ export default class ClientInstanceStore implements IClientInstanceStore {
         return rows.map((r) => r.app_name);
     }
 
+    async getDistinctApplicationsCount(daysBefore?: number): Promise<number> {
+        let query = this.db.from(TABLE);
+        if (daysBefore) {
+            const after = new Date();
+            after.setDate(after.getDate() - daysBefore);
```

Comment:
`subDays(after, daysBefore)` from date-fns? just to be absolutely clear? I see it works surprisingly well with setDate as well, I didn't think `setDate(-1)` would be a good call, but it does what  you hope for, second to last day in previous month.

Reply:
I like the subDays, what I don't like from my solution is that you need to mutate a date to get the result

Reply:
```typescript
test('subdays', () => {
    const theDate = new Date();
    const minus3 = (d: Date) => {
        d.setDate(d.getDate() - 3);
        return d;
    };
    expect(subDays(theDate, 3)).toEqual(minus3(theDate));
});
```
Just double checked this locally (but I don't think I need to add this test)